app/src/main/java/com/example/babytracking/BreastActivity.kt

```diff
@@ -1,11 +1,136 @@
+    }
+
+    private fun breastTimeClick() {
+        val timePicker = findViewById<TimePicker>(R.id.sleep_timepick_hour)
+        timePicker.setOnTimeChangedListener { _, pickerHour, pickerMinute ->
+            hour = pickerHour
+            minute = pickerMinute
+
+            // AM_PM decider logic
+            when {
+                hour == 0 -> {
+                    hour += 12
+                    ampm = "AM"
+                }
+                hour == 12 -> ampm = "PM"
+                hour > 12 -> {
+                    hour -= 12
+                    ampm = "PM"
+                }
+                else -> ampm = "AM"
+            }
+        }
+    }
+
+    private fun breastSelection() {
+        if (spinner_breast != null) {
+            val breastSpinnerAdapter = ArrayAdapter.createFromResource(
+                this, R.array.array_breast,
+                android.R.layout.simple_spinner_item
+            )
+
+            breastSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+            spinner_breast.adapter = breastSpinnerAdapter
+
+
+            spinner_breast.onItemSelectedListener =
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onItemSelected(
+                        parent: AdapterView<*>, view: View, pos: Int, id: Long
+                    ) {
+                        if (! TextUtils.isEmpty(parent.getItemAtPosition(pos) as String)) {
+                            breast = spinner_breast.selectedItem.toString()
+                        }
+                    }
+
+                    override fun onNothingSelected(parent: AdapterView<*>?) {
+                        breast = getString(R.string.unknown_breast)
+                    }
+                }
+        }
+    }
+
+    private fun breastSave() {
+        breast_button_save.setOnClickListener {
+
+            val lengthText: String = breast_length.text.toString().trim()
+            val lenghtInt: Int = Integer.parseInt(lengthText)
+            val babyText = breast_quality_baby.text.toString().trim()
+            val motherText = breast_quality_mother.text.toString().trim()
+            val hourVal = if (hour < 10) "0" + hour else hour
+            val minVal = if (minute < 10) "0" + minute else minute
+            val timeText = "$hourVal : $minVal $ampm"
+
+            if (lengthText.isEmpty()) {
+                breast_length.error = "Length of breastfeeding required"
+                breast_length.requestFocus()
+                return@setOnClickListener
+            } else if (lenghtInt <= 0) {
+                breast_length.error = "Time cannot be negative or null"
+                breast_length.requestFocus()
+                return@setOnClickListener
+            } else if (lenghtInt > 60) {
+                breast_length.error = "Time cannot exceed 60 minutes"
+                breast_length.requestFocus()
+                return@setOnClickListener
+            }
```

Comment:
Would these actually ever happen given that the time picker doesn't allow that? I think it's fine to check for it, but having an error message might be overkill since it's unlikely to happen